static/src/javascripts/projects/common/modules/identity/upsell/button/FollowButtonWrap.js

```diff
@@ -0,0 +1,132 @@
+// @flow
+import React, { Component } from 'preact-compat';
+import envelopeRemove from 'svgs/icon/envelope-remove.svg';
+import envelopeAdd from 'svgs/icon/envelope-add.svg';
+import tick from 'svgs/icon/tick.svg';
+
+const getButtonClassNames = (...modifiers: string[]) => [
+    'manage-account__button',
+    ...modifiers.map(m => `manage-account__button--${m}`),
+];
+
+type FollowButtonWrapProps = {
+    following: boolean,
+    onFollow: () => Promise<void>,
+    onUnfollow: () => Promise<void>,
+    trackingName: ?string,
+};
+
+type FollowButtonWrapState = {
+    mousedOutOnce: boolean,
+};
+
+class FollowButtonWrap extends Component<
+    FollowButtonWrapProps,
+    FollowButtonWrapState
+> {
+    constructor(props: FollowButtonWrapProps) {
+        super(props);
+        this.setState({
+            mousedOutOnce: true,
+        });
+    }
+
+    onMouseOut = () => {
+        this.setState({
+            mousedOutOnce: true,
+        });
+    };
+
+    updateFollowing = (to: boolean) => {
+        if (to) {
+            this.props.onFollow();
+        } else {
+            this.props.onUnfollow();
+        }
+        this.setState({
+            mousedOutOnce: false,
+        });
+    };
+
+    render() {
+        const { following, trackingName } = this.props;
+        const { mousedOutOnce } = this.state;
+        return following ? (
```

Comment:
It might be against convention but I find this ternary quite hard to read - could it be broken up at all for readability?

Reply:
will turn it into 2 variables at the very least